src/org/sosy_lab/java_smt/basicimpl/IndependentInterpolatingProverEnvironment.java

```diff
@@ -127,17 +124,41 @@ public BooleanFormula getInterpolant(Collection<TFormulaInfo> pFormulasOfA)
     Collection<BooleanFormula> formulasOfA = formulasAAndB.gotFormulasForA();
     Collection<BooleanFormula> formulasOfB = formulasAAndB.gotFormulasForB();
 
-    if (formulasOfA.isEmpty() || formulasOfB.isEmpty()) {
-      throw new IllegalArgumentException(
-          "Independent interpolation cannot be done over empty interpolation groups.");
+    BooleanFormula conjugatedA = bmgr.and(formulasOfA);
+    BooleanFormula conjugatedB = bmgr.and(formulasOfB);
+
+    // handle empty interpolation groups and trivial interpolants
+    if (formulasOfA.isEmpty()) {
+      conjugatedA =  bmgr.makeTrue();
+    }
+    if (formulasOfB.isEmpty()) {
+      conjugatedB = bmgr.makeTrue();
+    }
+
+    if ((bmgr.isTrue(conjugatedA) && bmgr.isTrue(conjugatedB))) {
+      return null;
```

Comment:
You should never return `null` for interpolating. If you can`t determine a correct value you should throw an exception.